app/src/main/java/org/piramalswasthya/sakhi/configuration/BenRegFormDataset.kt

```
@@ -2058,5 +2259,32 @@ class BenRegFormDataset(context: Context, language: Languages) : Dataset(context
     }
+    private fun shouldShowMaternalDeath(gender: String?, dob: String?): Boolean {
+        Log.v("valuesOfData","dob:$dob")
+
+        if (gender != "Female") return false
+        val age = dob?.let { dobString ->
+            val sdf = SimpleDateFormat("dd-MM-yyyy", Locale.getDefault())
+            try {
+                val date = sdf.parse(dobString)
+                val dobInMillis = date?.time ?: return false
+                getAgeFromDob(dobInMillis)
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)